### PR TITLE
middleware to return all entities with view permission

### PR DESCRIPTION
**Pull request by mateen-hussain**

### CODE REVIEW

File: helpers/entity.js [hunk 1]
```diff
@@ -136,6 +136,45 @@ class EntityHelper {
     await this.buildHierarchy(entity, parents)
     return parents;
   }
+  
+  async entitiesWithViewPermission(roles) {
+    if(!this.options.roles) {
```
Comment (leighjohnson-friday) [inline helpers/entity.js:141]: It doesn't appear you use this.options.roles in this function.  Should this be checking the roles function param instead?

File: middleware/entityUserPermissions.js [hunk 2]
```diff
@@ -17,10 +16,9 @@ async function entitiesUserCanAccess(user) {
         where: { userId: user.id }
       }
     });
-
-    whitelist = await entityHelper.entitiesWithRoles(roles);
```
Comment (leighjohnson-friday) [inline middleware/entityUserPermissions.js:21]: I believe there is another instance in the code where entitiesWithRoles is used.  This should also be updated.

File: helpers/entity.js [hunk 3]
```diff
@@ -136,6 +136,45 @@ class EntityHelper {
...
+      where: {
+        roleId: roleIds,
+        canEdit: false
+      } });
+    let entitesWithViewRoles = {};
+    for(const vr of viewRoleEntities){
+      entitesWithViewRoles[vr.entityId] = es[vr.entityId];
+      if(vr.shouldCascade && es[vr.entityId].children.length >0) {
+        const childIds = es[vr.entityId].children.map(c => c.id);
+        for(const cid of childIds) {
```
Comment (leighjohnson-friday) [inline helpers/entity.js:156]: Do you need the map before looping through the children.  Could you just use  for(const cid of es[vr.entityId].children) { instead?

Comment (mateen-hussain) [reply helpers/entity.js:156]: agreed.

File: middleware/entityUserPermissions.js (hunk 2 above)
Comment (mateen-hussain) [reply middleware/entityUserPermissions.js:21]: I wasnt aware of updating, all instances. WIll update it and delete that function then ?

**Review by leighjohnson-friday — APPROVED** [review]

**Pull request merged**